components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2858,26 +2858,29 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             Permission.ADMIN_WORKSPACES,
         );
 
-        await this.workspaceDb.trace(ctx).transaction(async (db) => {
```

Comment:
this transaction was unnecessary. but since I added the `addWorkspaceToOrg` call here so that we add relationships when an admin is restoring a softdeleted workspaces, I wrapped just the two write operations and added a rollback logic as we usually do.

Reply:
I don't think so: We do check properties on the Workspace, alter it based on the outcome, and write the _whole shape_ back to the DB. These kind of changes should ideally always be wrapped.

I this case, though, where it's more of a "conditional write", it IMO would be fine to leave the current version, if we `db.store(ws)` to `db.updatePartial(ws.id, { pinned, softDeletedTime, softDeleted })`. :+1:

Reply:
there is no locking or so involved with transactions. We read state from DB and use that in memory and then write it back once. it will overwrite whatever is in the DB at that point no matter there was a long running transaction or not.

Reply:
> there is no locking or so involved with transactions.

You are right, getting ahead of myself here :see_no_evil: 

> it will overwrite whatever is in the DB

:+1: `updatePartial` would limit the blast radius in a way, then. :+1: